scipy/stats/_stats_py.py

```diff
@@ -6134,27 +6223,82 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate',
     uniform distribution, which *does* have a population mean of 0.5, we would
     mistakenly reject the null hypothesis for one of them.
 
-    """
-    a, axis = _chk_asarray(a, axis)
+    `ttest_1samp` can also compute a confidence interval around the population
+    mean.
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    >>> rvs = stats.norm.rvs(size=50, random_state=rng)
+    >>> res = stats.ttest_1samp(rvs, popmean=0)
+    >>> ci = res.confidence_interval(confidence_level=0.95)
+    >>> ci
+    ConfidenceInterval(low=-0.3193887540880017, high=0.2898583388980972)
+
+    The bounds of the 95% confidence interval are the
+    minimum and maximum values of the parameter `popmean` for which the
+    p-value of the test would be 0.05.
+
+    >>> res = stats.ttest_1samp(rvs, popmean=ci.low)
+    >>> np.testing.assert_allclose(res.pvalue, 0.05)
+    >>> res = stats.ttest_1samp(rvs, popmean=ci.high)
+    >>> np.testing.assert_allclose(res.pvalue, 0.05)
+
+    Under certain assumptions about the population from which a sample
+    is drawn, the confidence interval with confidence level 95% is expected
+    to contain the true population mean in 95% of sample replications.
+
+    >>> rvs = stats.norm.rvs(size=(50, 1000), loc=1, random_state=rng)
+    >>> res = stats.ttest_1samp(rvs, popmean=0)
+    >>> ci = res.confidence_interval()
+    >>> contains_pop_mean = (ci.low < 1) & (ci.high > 1)
+    >>> contains_pop_mean.sum()
+    953
 
-    if contains_nan and nan_policy == 'omit':
-        a = ma.masked_invalid(a)
-        return mstats_basic.ttest_1samp(a, popmean, axis, alternative)
+    """
+    a, axis = _chk_asarray(a, axis)
 
     n = a.shape[axis]
     df = n - 1
 
-    d = np.mean(a, axis) - popmean
+    mean = np.mean(a, axis)
+    d = mean - popmean[..., 0]  # popmean is an array because of decorator
```

Comment:
Hmm the behavior was buggy before (see https://github.com/scipy/scipy/pull/16835/files#r945519181) but  we still have a problem with this code: if the user were to pass `popmean` of the same shape as `a`, for instance, then only the 0th element along `axis` would be considered.

Reply:
I think the right thing to do is to remove (only) the `axis` dimension of `popmean` (only) if it's a singleton, but I'd need to play with that.

Reply:
```suggestion
    popmean = np.squeeze(popmean, axis=axis)
    d = mean - popmean
```
This is a better fix. It does the right thing, and it doesn't silently ignore the case in which `popmean` has multiple values along `axis` (which is always -1 because of the decorator).

We have a choice here. We could do `mean = np.mean(a, axis, keepdims=True)`, in which case this would work even if `popmean.shape[axis] !=1`; that is, the user could test each sample in `a` against multiple hypothesized `popmean` values by placing those values of `popmean` along the indicated `axis`.

I don't think we should allow that. We would have a funny case here in which `popmean` wants to do an element-wise operation and `a` wants a reduction operation. Instead, I think we should think of `popmean` as a sample that must have size 1, and it also gets a reduction operation along `axis` (hence `squeeze`). It's equivalent to (but more efficient than) `popmean` being subtracted from every element of `a` before the mean is taken. 

The user can still test multiple null hypotheses per sample by giving `popmean` an axis orthogonal to those in `mean`, e.g.:

```python3
x = np.random.rand(1, 15, 20)
popmean = np.ones((5, 1, 20))  # test 5 alternative hypotheses for each of 20 samples of size 15
res = stats.ttest_1samp(x, popmean=popmean, axis=1)
res.statistic.shape  # (5, 20)
```
This strictly follows normal broadcasting rules, which I think is preferable to making up special cases. So I think we should raise an error if `popmean.shape[axis] != 1`.

```suggestion
    try:
        popmean = np.squeeze(popmean, axis=axis)
    except ValueError as e:  # when axis is not a singleton
        raise ValueError("`popmean.shape[axis]` must equal 1.") from e
    d = mean - popmean
```